liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+            self._write([0x58, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'fixed':
+            self._write_color([colors[0], colors[0]])
+            self._write([0x55, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'alert':
+            temps = list(temps.split(' '))
+            LOGGER.debug('temps count = %d', len(temps))
+            LOGGER.debug('asetek temps %s %s %s', temps[0], temps[1], temps[2])
+            LOGGER.debug('color count = %d', len(colors))
+            self._write([0x5f, int(temps[0]), 0x00, int(temps[1]), 0x00, int(temps[2]), 0x00] + colors[0] + colors[1] + colors[2])
+            self._end_transaction_and_read(6)
+            self._write([0x5E, 0x01])
+            self._end_transaction_and_read(3)
+        else:
+            raise KeyError('Unknown lighting mode {}'.format(mode))
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to follow a speed duty profile."""
+        if channel == 'pump':
+            raise ValueError('set_speed_profile not implemented for the pump on CorsairAsetekPro')
+        adjusted = self._prepare_profile(profile, 0, 100, _MAX_PROFILE_POINTS)
+        for temp, duty in adjusted:
+            LOGGER.info('setting %s PWM point: (%i°C, %i%%), device interpolated',
+                        channel, temp, duty)
+        temps, duties = map(list, zip(*adjusted))
+        # Need to write curve for each fan
+        # Assume if we read n values from _get_fan_speed the fans are numbered [0, n)
+        # This will have problems on a 2 fan system when the first fan has failed
+        for i in self._get_fan_indexes(channel):
+            self._write([_PRO_CMD_WRITE_FAN_SPEED, i] + temps + duties)
+            self._end_transaction_and_read(32)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set channel to a fixed speed duty."""
+        if channel.startswith('fan'):
+            # Setting a fix speed is supported, however it is done by sending RPM or PWM
+            # values. I'm not 100% sure how this maps to %s used by liquidclt so set a flat
+            # speed profile instead.
+            LOGGER.info('using a flat profile to set %s to a fixed duty', channel)
+            self.set_speed_profile(channel, [(0, duty), (_CRITICAL_TEMPERATURE - 1, duty)])
+            return
+        # Fixed speed pump is not supported, so guess a curve from the duty value given
+        # less than 50 == quiet (0), less than 80 == balanced (1), greater than 80 == proformance (2).
+        # Arbitrary values are arbitrary.
+        if duty < 50:
+            duty = 0
+        elif duty < 80:
+            duty = 1
+        else:
+            duty = 2
+        self._write([_PRO_CMD_WRITE_PUMP_MODE, duty])
+        self._end_transaction_and_read(5)
+        # OCL Reads the mode after setting it, so do the same
+        self._write([_PRO_CMD_READ_PUMP_MODE])
+        self._end_transaction_and_read(4)
```

Comment:
I think this is dead code? it is after a return statement

Reply:
That return statement is only met if channel is 'fan1', 'fan2', etc.
It's a if condition for 'fan' or 'pump' mode when changing speeds.

Reply:
My bad, the side by side diff made  it look like the `return` was not indented.